src/main/groovy/org/shipkit/internal/gradle/git/GitCheckOutTask.java

```diff
@@ -27,8 +36,45 @@ public String getRev() {
         return rev;
     }
 
+    /**
+     * See {@link #isNewBranch()}
+     */
+    public void setNewBranch(boolean newBranch) {
+        this.newBranch = newBranch;
+    }
+
+    /**
+     * Should checkout create a new branch
+     */
+    public boolean isNewBranch() {
+        return newBranch;
+    }
+
     @TaskAction
     public void checkOut() {
-        new DefaultProcessRunner(getProject().getProjectDir()).run("git", "checkout", rev);
+        getProcessRunner().run(getCommandLine());
+    }
+
+    private List<String> getCommandLine(){
+        List<String> commandLine = new ArrayList<String>();
+        commandLine.add("git");
+        commandLine.add("checkout");
+        if(newBranch){
+            commandLine.add("-b");
+        }
+        commandLine.add(rev);
+        return commandLine;
+    }
+
+    private ProcessRunner getProcessRunner(){
+        if(processRunner == null){
+            return new DefaultProcessRunner(getProject().getProjectDir());
+        }
+        return processRunner;
+    }
+
+    @ExposedForTesting
+    protected void setProcessRunner(ProcessRunner processRunner){
```

Comment:
It's not so easy with Gradle tasks, they cannot be instantiated directly (they throw TaskInstantiationException if you try to call one of the constructors). See constructors of org.gradle.api.internal.AbstractTask for more details. I didn't find a better way of testing than the one above. If you find something better, please let me know.